### ISSUE TITLE
Custom label?

### ISSUE THREAD

**Issue by hopewise** [kristijanhusak/laravel-form-builder#295]
As I've read the docs, its seems that the label is the same as the filed name, with Capital starting letter, however, in some cases, we might need to customize the label, ex: translation.

How would we achieve this?

**Comment by kristijanhusak** [comment]
There are two ways to tackle this:

1.  just pass the `label` property to the field.

```php
$this->add('name', 'text', [
  'label' => trans('first_name')
])
```

2. Add `language_name` to the form class, and it will search for translation in there automatically. Just make sure the field name exists in the translations to be properly translated. If it's not found, it defaults to ucfirst() field name.
```php
class MyForm extends Form {
  protected $languageName = 'translations.forms.my_form' // lang/en/translations.php. content below

  public function buildForm()
  {
    $this->add('name', 'text');
  }
}
```

```php
// lang/en/translations.php

return [
  'forms' => [
    'my_form' => [
      'name' => 'First name'
    ]
  ]
];
```

**Comment by CsFender** [comment]
Hi!

I tried this setting, but it doesn't work on 'Entity' labels. :(
Can you check this please? Thanks a lot!

`$languageName = 'translations.forms.my_form'`

**Comment by rudiedirkx** [comment]
Works fine for me. All field types use the same label mechanism (`FormField::setupLabel()`), so every field type should work. Debug `setupLabel()` or `FormHelper::formatLabel()` to find out why it's not working.

**Comment by rudiedirkx** [comment]
Or do you mean the `<option>` labels...? Those come from your db, not the translator.

**Comment by CsFender** [comment]
I mean to the "admin_user" text.

`<div class="choice-wrapper">
<input id="permissions_admin_user" name="permissions[]" type="checkbox" value="admin_user"> 
<label for="permissions_admin_user" class="choice-label-class">admin_user</label>
</div>`

**Comment by CsFender** [comment]
I mean to the "admin_user" text.

![er](https://user-images.githubusercontent.com/8111835/56173160-d3180c80-5fec-11e9-821b-c763fad3823c.jpg)

**Comment by rudiedirkx** [comment]
Right, that's the `<option>` (if not multiple) I was talking about. That's not translated. You have to do that manually by making the entity option `property` a callback, which returns a single entity's label:

```
$this->add('permissions', 'entity', [
    'class' => Permission::class,
    'property' => function(Permission $permission) {
        return trans("foo.bar.$permission->trans_key");
    },
]);
```
